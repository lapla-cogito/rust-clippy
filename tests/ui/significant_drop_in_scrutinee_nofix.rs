#![warn(clippy::significant_drop_in_scrutinee)]
//@no-rustfix

use std::sync::Mutex;

fn should_trigger_lint_in_while_let() {
    let mutex = Mutex::new(vec![1]);

    while let Some(val) = mutex.lock().unwrap().pop() {
        //~^ ERROR: temporary with significant `Drop` in `while let` scrutinee will live until the
        //~| NOTE: this might lead to deadlocks or other unexpected behavior
        println!("{}", val);
    }
}
