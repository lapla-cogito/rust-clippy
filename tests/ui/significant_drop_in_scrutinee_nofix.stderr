error: temporary with significant `Drop` in `while let` scrutinee will live until the end of the `while let` expression
  --> tests/ui/significant_drop_in_scrutinee_nofix.rs:9:27
   |
LL |     while let Some(val) = mutex.lock().unwrap().pop() {
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
LL |     }
   |      - temporary lives until here
   |
   = note: this might lead to deadlocks or other unexpected behavior
   = note: `-D clippy::significant-drop-in-scrutinee` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::significant_drop_in_scrutinee)]`

error: aborting due to 1 previous error

